intake_dal/dal_catalog.py

```diff
@@ -68,3 +81,8 @@ def _set_dal_default_storage_mode(self, data):
                     v["args"]["default"] = self.storage_mode
                 elif isinstance(v, dict):
                     self._set_dal_default_storage_mode(v)
+
+    @staticmethod
```

Comment:
I am curious and this maybe outside the scope of this PR so please pardon, can the input datasets be in any format, for example for feature transformations we have datasets living in parquet and possibly in the future zarr: https://zarr.readthedocs.io/en/stable/ . how do we reconcile avro with these formats

Reply:
The idea of intake-dal (and Intake) is it should support any dataformat.

Avro can describe multi-dimensional array schema.  Getting the avro_schema for datasource that doesn't support it would return `None`.